### DISCUSSION TITLE
Display PathBuf as String

### DISCUSSION THREAD

**Discussion by oneElectron** [SeaQL/FireDBG.for.Rust#8] [Ideas]
I know that it's early days for the project, and I don't know how difficult this may be to implement.

# Use Case
I have a project that works a lot with the file system.
For my project it would be nice if FireDBG displayed PathBufs as strings, instead of as the underlying bytes.

# Some Concerns
I am aware that this would require some discussion on when FireDBG should decide to stringify a type.
I am also aware that this might be somewhat difficult, given that PathBufs are stored as OsStrings, which means they might not be valid UTF-8 like normal Strings.

**Comment by tyt2y3** [comment]
I think this is doable. We always serialize strings as bytes, and only decode them as utf8 in indexer. So if it can't decode properly we can display it as bytes.

**Reply by oneElectron** [reply]
Sounds great!

**Reply by tyt2y3** [reply]
Released an update. Should now render as string if parsable as utf8, and bytes otherwise.

![image](https://github.com/SeaQL/FireDBG.for.Rust/assets/1782664/27866205-6062-43d7-882f-4c4f74e00a6a)

**Reply by oneElectron** [reply]
Nice! Works perfectly for me. Thanks
